src/hooks/api/useSplits.ts

```diff
@@ -1,34 +1,134 @@
 import {
+  keepPreviousData,
   useQuery,
   UseQueryResult,
 } from '@tanstack/react-query';
 import type { FindOptionsWhere } from 'typeorm';
 
 import { Split } from '@/book/entities';
 import type { Account } from '@/book/entities';
-import { getSplits } from '@/lib/queries';
 import fetcher from './fetcher';
 
 /**
- * This query is mainly used for when we want to retrieve splits for a specific account
- * so we display them
+ * This query is a generic way to retrieve splits. It's quite unoptimised because
+ * it is used by useInvestment/s queries and they need A LOT of nested information
+ * like full transation with all splits, the account, etc. We probably should separate
+ * in two queries in the future.
  */
-export function useSplits(findOptions: FindOptionsWhere<Account>): UseQueryResult<Split[]> {
-  return useQuery({
-    queryKey: [...Split.CACHE_KEY, findOptions],
+export function useSplits(
+  findOptions: FindOptionsWhere<Account>,
+): UseQueryResult<Split[]> {
```

Comment:
few changes here, we now have 4 hooks in useSplits:

- useSplits: generic one, allows us to query all splits for an account, all splits for a specific account type, etc.
- useSplitsPagination: retrieve splits for an account with pagination
- useSplitsCount: get total number of splits for an account (useful for pagination to know number of pages).
- useSplitsTotal: get the total quantity for the splits of an account (useful for counting totals).